### ISSUE TITLE
Trace command do not work !

### ISSUE THREAD

**Issue by stringhuang** [emqx/emqx#2385]
#### Environment

- OS: CentOS Linux release 7.4.1708 (Core)
- Erlang/OTP: 20
- EMQ: 3.0.1
- Node: uds-gww1, uds-gww2

#### Description

Trace functionality do not take effect !

At 2019-03-29 17:47, I start the tracing for client daluox on both nodes.

![image](https://user-images.githubusercontent.com/44996282/55224266-1a5f7880-524b-11e9-9f76-55330455eff2.png)

![image](https://user-images.githubusercontent.com/44996282/55224297-29dec180-524b-11e9-8278-205681fdee9b.png)

At 2019-03-29 17:52:06, I published message 'asdf' to topic 'xxx' by using clientId 'daluox', which is succeeded by consumed.

![image](https://user-images.githubusercontent.com/44996282/55224407-6ca09980-524b-11e9-9e33-781db6cef9e7.png)

![image](https://user-images.githubusercontent.com/44996282/55224416-74f8d480-524b-11e9-875c-1e08c6ceb162.png)

However, the trace file is still empty on both nodes.

![image](https://user-images.githubusercontent.com/44996282/55224521-ac678100-524b-11e9-8ff6-73abb1f48b07.png)

![image](https://user-images.githubusercontent.com/44996282/55224540-b7221600-524b-11e9-9fec-68879ccff5aa.png)

By the docs "https://developer.emqx.io/docs/emq/v3/cn/commands.html?highlight=trace#trace" said, trace command will store all the log of specific client/topic into the file. However, from my test case, it do not ! please help !

**Comment by phanimahesh** [comment]
Can you check the log level and log handlers after starting trace?

**Comment by stringhuang** [comment]
Hi @phanimahesh , from my capture "bin/emqx_ctl trace list", we may see the log level is "ALL", so that it should be okay to keep trace. And what do you mean by "log handlers"?

**Comment by terry-xiaoyu** [comment]
@stringhuang Hi, it is our doc that is not clear about the log level, we're working on fixing the docs.

Before using the log trace feature, you should set the `primary logger level` to a lower enough value (the default value is `error`), by `emqx_ctl log` :

```shell
./bin/emqx_ctl log primary-level debug

./bin/emqx_ctl trace start client 'client-1' /tmp/trace-1.log debug
```

Hope this helps. I'll improve the CLI so that it won't accept the trace command if the current primary level is too high.

**Comment by stringhuang** [comment]
Hi @terry-xiaoyu , Thanks for your information. 
After I set `primary logger level` as `debug`, I now can succeeded view the log of one specific client. 
From the capture, the file `/root/daluo0405x09.log` for tracing client daluo0405x09 has some content there, but the file `/root/ttx09.log` is still empty, though either I have published to or subscribed from the topic `ttx09`. So, is there another setting to trace one specific topic ? 

![image](https://user-images.githubusercontent.com/44996282/55703356-fa2e7700-5a0b-11e9-879f-4d4db590dedd.png)

**Comment by terry-xiaoyu** [comment]
No, there's no other configurations, it should work. Please check if the message is published to the right topic (`ttx09`).

BTW, the topic tracing only works on publish messages, not on subscribe messages.

**Comment by stringhuang** [comment]
Hi @terry-xiaoyu, since my emqx env is under cluster with two nodes, so I started trace topic on both nodes, and it now can succeed trace the specific topic of published message from which clientId, IP, port, etc, which is kept only on one node for one message. Before I just start trace only on one node, eg. nodeA, so it may miss info if the publish client connect to the nodeB. is this the case ?

**Comment by terry-xiaoyu** [comment]
The current cluster joining mechanism will restart the newly joined node, so all the runtime configs will be lost, including the tracing log handler. And the tracing is now only applied on a single node, so you have to execute the trace command on BOTH nodes AFTER the cluster has been established. 

We're thinking about creating a GUI in emqx dashboard or somewhere, which can tracing clients/topics   from all nodes and show the results on the web page. Actually it is a kind of `log collection`. But I don't know whether there's such a requirement.

**Comment by HJianBo** [comment]
Hi guys, This fix has released at v3.2.rc.2
